src/rvi.c

```diff
@@ -1562,11 +1562,9 @@ int rviProcessInput(TRviHandle handle, int *fdArr, int fdLen)
 
         json_decref( root );
     }
-        
-    free( buf );
-
 
 exit:
+    if( buf ) free( buf );
```

Comment:
FYI, free on a null pointer has no effect (i.e. the if statement is superfluous).